### PR TITLE
Fix #262 support access token

### LINKED ISSUE #262: Support access tokens for authentication
Artifactory Projects requires a valid token to authenticate.

https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-PROJECTS
https://www.jfrog.com/confluence/display/JFROG/Access+Tokens

### PR DESCRIPTION
**Pull request by zhan9san**
Fix #262 

Could you please review and approve this PR?

### CODE REVIEW

File: dohq_artifactory/auth.py [hunk 1]
```diff
@@ -4,15 +4,26 @@
 class XJFrogArtApiAuth(AuthBase):
     """Attaches X-JFrog-Art-Api Authentication to the given Request object."""
 
-    def __init__(self, apikey):
+    def __init__(self, apikey=None, token=None):
+        if apikey and token:
+            raise RuntimeError("Only one authentication method but get 2")
```
Comment (allburov) [inline dohq_artifactory/auth.py:9]: Can we clarify it a little?
`Use either token or apikey for authentication"

File: dohq_artifactory/auth.py [hunk 2]
```diff
@@ -4,15 +4,26 @@
...
+        )
 
     def __ne__(self, other):
         return not self == other
 
     def __call__(self, r):
-        r.headers["X-JFrog-Art-Api"] = self.apikey
+        if self.apikey:
+            r.headers["X-JFrog-Art-Api"] = self.apikey
+        else:
```
Comment (allburov) [inline dohq_artifactory/auth.py:27]: Shouldn't we add one more `if self.token`? 
Right now it looks like we always will add `Authorization` header, even if we don't define it. Just to make it more clear and readable

File: artifactory.py [hunk 3]
```diff
@@ -1248,13 +1248,18 @@ def __new__(cls, *args, **kwargs):
 
         # Auth section
         apikey = kwargs.get("apikey")
+        token = kwargs.get("token")
         auth_type = kwargs.get("auth_type")
-        if apikey is None:
+
+        if apikey:
+            logging.debug("Use XJFrogApiAuth apikey")
+            obj.auth = XJFrogArtApiAuth(apikey=apikey)
```
Comment (allburov) [inline artifactory.py:1256]: There're too many if - we have `if apikey \ if token` here and `if apikey \ if token` in `XJFrogArtApiAuth`.

Let's create a new class for TokenBased authentication in `auth.py`
```python
class XJFrogArtApiAuth:
    """Authenticaiton via ApiKey"""
    .... # Use X-JFrog-Art-Api header
class XJFrogArtBearerAuth:
   ... # Use Authorization Bearer header
```

In this case we'll have if only here when we choose one of authentication method.

Comment (zhan9san) [reply artifactory.py:1265]: A new class `XJFrogArtBearerAuth` is added, could you please review it again?

File: dohq_artifactory/auth.py (hunk 2 above)
Comment (zhan9san) [reply dohq_artifactory/auth.py]: Now, we have a new class for token, and this issue is gone as well.

File: dohq_artifactory/auth.py [hunk 4]
```diff
@@ -16,3 +16,20 @@ def __ne__(self, other):
...
+
+
+class XJFrogArtBearerAuth(AuthBase):
+    """Attaches X-JFrog-Art-Bearer Authentication to the given Request object."""
+
+    def __init__(self, token):
+        self.token = token
+
+    def __eq__(self, other):
+        return all([self.token == getattr(other, "token", None)])
```
Comment (allburov) [inline dohq_artifactory/auth.py:28]: We don't need `all` now

Comment (allburov) [reply dohq_artifactory/auth.py:28]: Let's keep it

**Pull request merged**